api/scanner/media_encoding/executable_worker/magickwand.go

```
@@ -90,8 +90,8 @@ func (cli *MagickWand) GenerateThumbnail(inputPath string, outputPath string, wi
 		return fmt.Errorf("ImagickWand set orientation for %q error: %w", inputPath, err)
 	}
 
-	// If the original image is rotated by 90 degrees, swap width and height for thumbnail generation
-	if originalWidth != wand.GetImageWidth() && originalHeight != wand.GetImageHeight() {
+	// If the image was rotated by 90 degrees (dimensions swapped), adjust thumbnail dimensions
```

I prefer current way. When I read it, it clearly says `when "width != width, height != height" then swap them`. I need to focus it a while then notice the new way is `width == height, height == width`.

---

Yeah, the current way looks more natural; however, it might work incorrectly and hide potential errors:
Imagine, image dimensions somehow corrupted in a previous step, and now they don't match the original ones. With the current implementation, they will be swapped, but it is wrong. By setting this `if` verification, we want to detect the exact case of 90-degree image rotation, and to serve this purpose, the `width == height, height == width` condition is better. All other cases of dimensions mismatch should not go inside this `if`, as they don't mean the 90-degree rotation.